test/SILGen/enum_raw_representable_available.swift

```diff
@@ -0,0 +1,67 @@
+// RUN: %target-typecheck-verify-swift
+// RUN: %target-swift-emit-silgen -emit-sorted-sil %s | %FileCheck %s
+// RUN: %target-swift-emit-silgen -emit-sorted-sil -enable-resilience %s | %FileCheck %s
+
+// Really just requires a platform with meaningful runtime #available() tests.
+// REQUIRES: objc_interop
+
+public enum E: Int {
+  // For some reason, we generate strange SIL for the first case that's
+  // difficult to validate. This just gets that out of the way.
+  case sacrificial = -500
+
+  case normal = -1000
+  @available(macOS 10.8, iOS 7, watchOS 1, tvOS 7, *)
+  case alwaysAvailable = -2000
+  @available(macOS 500.600.700, iOS 500.600.700, watchOS 500.600.700, tvOS 500.600.700, *)
+  case potentiallyUnavailable = -3000
+  @available(macOS, unavailable) @available(iOS, unavailable)
+  @available(watchOS, unavailable)  @available(tvOS, unavailable)
+  case neverAvailable = -4000
+}
+
+// CHECK-LABEL: sil {{(\[serialized\] )?}}[ossa] @$s32enum_raw_representable_available1EO0B5ValueACSgSi_tcfC
+
+// CHECK-NOT: integer_literal $Builtin.IntLiteral, -4000
```

Comment:
Rather than insert this everywhere, I suggest just doing two passes with FileCheck. (You can see examples of other tests doing this with `-check-prefix NEGATIVE`.)